cypress/utils/commands.osd.js

```diff
@@ -291,6 +291,41 @@ cy.osd.add('grabDataSourceId', (workspaceName, dataSourceName) => {
   }
 });
 
+cy.osd.add('grabIdsFromDiscoverPageUrl', () => {
+  cy.url().then(($url) => {
+    const workspaceIdMatch = $url.match(/\/w\/([^\/]+)\//);
+    const datasourceIdMatch = $url.match(/dataSource:\(id:'?([0-9a-f-]+)'?,/);
+    const indexPatternIdMatch = $url.match(/\),id:'?([0-9A-Za-z:_-]+)'?,/);
+
+    if (workspaceIdMatch && workspaceIdMatch[1]) {
+      cy.wrap(workspaceIdMatch[1]).as('WORKSPACE_ID');
+    }
+    if (datasourceIdMatch && datasourceIdMatch[1]) {
+      cy.wrap(datasourceIdMatch[1]).as('DATASOURCE_ID');
+    }
+    if (indexPatternIdMatch && indexPatternIdMatch[1]) {
+      cy.wrap(indexPatternIdMatch[1]).as('INDEX_PATTERN_ID');
+    }
+  });
+});
+
+cy.osd.add('deleteAllSavedSearches', (workspaceName) => {
```

Comment:
Could we skip the UI here and just use the saved objects API?

Reply:
implemented it below using @kavilla 's recommendation